privacyidea/lib/utils/compare.py

```diff
@@ -97,7 +97,13 @@ def _compare_matches(left, comparator, right):
     :return: True or False
     """
     try:
-        return re.match("^" + right + "$", left) is not None
+        # check for regex modes
+        m = re.match(r'^(\(\?[a-zA-Z]+\))(.+)$', right)
+        if m and len(m.groups()) == 2:
```

Comment:
Why is this special case? Would we get different results?

Reply:
We support [Regex Modifiers](https://regex.programmingpedia.net/en/tutorial/5138/regex-modifiers-flags-) which need to be put at the beginning of the pattern (before the `"^"`).